vignettes/shared_cache.Rmd

```diff
@@ -8,45 +12,154 @@ vignette: >
+      ind_file = target_name,
+      input_ind_file = <ind>'1_data/out/compiled_data.rds.ind'</ind>,
+      gd_config = <data>'lib/cfg/gd_config.yml'</data>)
+</pre>
 
-Disadvantages of a shared cache (as currently implemented):
-* In a fast-paced collaborative development environment (e.g., a 'sprint'), it is challenging to maintain synchrony between the shared cache (the data) and the git repository (the metadata). Asynchrony is not a deal-breaker but does lead to more rebuilding than would be required for a slower-paced project.
-* Though we've done much to ensure this doesn't happen, it's conceivable that metadata will become corrupt relative to the data. Some monitoring and very occasional full rebuilding is recommended when practical.
-* Old files no longer referenced by the code can accumulate on the shared cache unless manually deleted. Though these will not interfere with ongoing analysis, they can take up storage space unnecessarily.
 
 ### Guidelines
 
-Projects using a shared cache should follow these guidelines:
+####Projects using a shared cache should follow these guidelines:
+
+* Use <ind>**indicator files**</ind> (usually an `.ind` suffix) to represent most or all of the chain of connected <target>**targets**</target> within your remake files. Each `.ind` file should be one of two products of a recipe (e.g. R script), where the other product is the creation of a <data>**data file**</data>, either locally and/or in the shared cache.
+
+* Always build <target>**targets**</target> using `scipiper::scmake()` rather than `remake::make()`. Though the functions are outwardly very similar, `scmake()` maintains an extra layer of metadata that allows multiple users to share a single project <build>**build status**</build> (e.g., "file x.rds.ind is up to date; file y.rds.ind is out of date"). In a shared-cache project, you should not even need to load the `remake` package directly.
+
+* Generally avoid using R objects as `remake/scmake` <target>**targets**</target>...but if you must, usually for convenience or conciseness of the workflow plan, recognize that R objects must be built by every analyst. So if a target takes a non-trivial length of time to build, or if it depends on large volumes of <data>**data**</data> as input, that target should usually be a file rather than an R object.
 
-* Use indicator files (usually an `.ind` suffix) to represent most or all of the chain of connected targets within your remake files. Each .ind file should be one of two products of a recipe, where the other product is the creation of a data file, either locally and/or in the shared cache.
+* `git commit` all <ind>**indicator files**</ind> and <build>**build files**</build> (with occasional exception of <ind>**indicator files**</ind> within a task plan; those require additional thought). `git ignore` all <data>**data files**</data> *except* those that are  necessary to pipeline basic functions (e.g. <data>**gd_config.yml**</data>). 
```

Comment:
I think the choice of which data files to commit is not so much about necessity but rather about whether they are amenable to storage in git/GitHub - relatively small, text-based, typically hand-curated, and having their canonical copy within the git repo rather than in the shared cache or another cloud location. For example, another data file we might commit could be a short .tsv file that maps lake IDs to the filenames of the corresponding collaborator-provided data files.

While for most data files I see git committing as optional, enabled but not required by the above criteria. In contrast, gd_config.yml really _should_ be git committed because it informs the basic pipeline function of data exchange with Google Drive...but I'd argue that gd_config.yml is only borderline a "data file" anyway, because its "data" are configuration parameters.

Reply:
I'll expand on the data file explanation for what can / should get committed. Thanks!